### PR TITLE
add s1meta.bursts as DA without geopandas at this stage

### PR DESCRIPTION
**Pull request by agrouaze**
address issue #54 
objective: add `s1meta.bursts` 

- [x] add mappings of annotations
- [x] add compound var
- [x] validation

This PR doesn't contain the method in `s1meta` allowing to get polygons defined in atrack/xtrack because it requires that PR #46 (s1meta.geoloc) to be available.

### CODE REVIEW

**Comment by agrouaze** [comment]
image

**Comment by agrouaze** [comment]
In https://github.com/umr-lops/xsar/pull/55/commits/3e73a59b66691b899920d30df93629a2d1df48d2 I removed attribute `lines_per_burst` since it is accessible via the `xarray.Dataset` returned y `s1meta.bursts` and depending on selections it is potentially preferable to avoid hard coded attribute.

File: src/xsar/sentinel1_meta.py [hunk 1]
```diff
@@ -831,6 +831,14 @@ def coords2heading(self, atracks, xtracks, to_grid=False, approx=True):
         _, heading = haversine(lon1, lat1, lon2, lat2)
         return heading
 
+    @property
+    def bursts(self):
+        if self.xml_parser.get_var(self.files['annotation'].iloc[0], 'annotation.number_of_bursts') > 0:
+            return self.xml_parser.get_compound_var(self.files['annotation'].iloc[0], 'bursts')
+        else:
+            # no burst, return empty dataframe with 'burst' column
+            return pd.DataFrame({'burst': []})
```
Comment (oarcher) [inline src/xsar/sentinel1_meta.py:840]: if `bursts` stucture is a xarray dataset, no bursts should also be a empty xarray

File: src/xsar/sentinel1_meta.py [hunk 2]
```diff
@@ -831,6 +831,14 @@ def coords2heading(self, atracks, xtracks, to_grid=False, approx=True):
         _, heading = haversine(lon1, lat1, lon2, lat2)
         return heading
 
+    @property
+    def bursts(self):
```
Comment (oarcher) [inline src/xsar/sentinel1_meta.py:835]: if bursts are not rasterisable, I think that we should name them as `self._bursts`, not `self.bursts`

**Review by oarcher — CHANGES_REQUESTED** [review]

File: src/xsar/sentinel1_meta.py (hunk 2 above)
Comment (oarcher) [reply src/xsar/sentinel1_meta.py]: solved by f024c7420

File: src/xsar/sentinel1_meta.py (hunk 1 above)
Comment (agrouaze) [reply src/xsar/sentinel1_meta.py]: DataFrame replaced by xarray.Dataset.
image
image

**Pull request merged**